web/src/reducers/activities.js

```diff
@@ -64,6 +77,29 @@ const reducer = (state = initialState, action) => {
         },
         state
       );
+    case ADD_ACTIVITY_EXPENSE:
+      return u(
+        {
+          byId: {
+            [action.id]: {
+              expenses: expenses => [...expenses, newExpense(expenses.length)]
```

Comment:
i don't think this will work -- let's say there are 3 expenses (0, 1, 2); if you delete the first one (0) and then add a new one, it will get an idx of 2 which will be a repeat.

3 options i think: do it similarly to `milestones` (no explicit id/idx field, just uses the index of the array element); create a `generateUniqueId` function and use that for new expenses; loop through existing expenses and get max idx and then add 1 to that.